gravitee-am-gateway/gravitee-am-gateway-services/gravitee-am-gateway-services-sync/src/main/java/io/gravitee/am/gateway/services/sync/SyncManager.java

```diff
@@ -112,19 +126,29 @@ public void refresh() {
             } else {
                 // search for events and compute them
                 logger.debug("Events synchronization");
-
-                List<Event> events = eventRepository.findByTimeFrame(lastRefreshAt - lastDelay, nextLastRefreshAt).toList().blockingGet();
-
-                if (events != null && !events.isEmpty()) {
-                    // Extract only the latest events by type and id
-                    Map<AbstractMap.SimpleEntry, Event> sortedEvents = events
-                            .stream()
-                            .collect(
-                                    toMap(
-                                            event -> new AbstractMap.SimpleEntry<>(event.getType(), event.getPayload().getId()),
-                                            event -> event, BinaryOperator.maxBy(comparing(Event::getCreatedAt)), LinkedHashMap::new));
-                    computeEvents(sortedEvents.values());
+                Completable eventProcessing = eventRepository.findByTimeFrame(lastRefreshAt - lastDelay, nextLastRefreshAt)
+                        .toList()
+                        .toMaybe()
```

Comment:
You either don't need this or you miss the filter part to replace following `if (!events.isEmpty()) {`